Tools/ardupilotwaf/boards.py

```diff
@@ -44,7 +44,7 @@ def configure(self, cfg):
                     keys.sort()
                 val = ['%s=%s' % (vk, val[vk]) for vk in keys]
 
-            if k in cfg.env and isinstance(cfg.env, list):
+            if k in cfg.env and isinstance(cfg.env.k, list):
```

Comment:
I believe this worked because `ConfigSet` evaluates a non-existing key to an empty list, so `isinstance(cfg.env.k, list)` is always true as long as `cfg.env` doesn't have the key `'k'`. That's why I check `k in cfg.env` before checking if `cfg.env[k]` is a list.